client/client.go

```diff
@@ -249,7 +257,6 @@ func NewClientWithOpts(ops ...func(*Client) error) (*Client, error) {
 	c := &Client{
 		host:    DefaultDockerHost,
 		version: api.DefaultVersion,
-		scheme:  "http",
```

Comment:
Should we be careful removing this here, in case there's an `ops` that expects this information to be set?

Reply:
> Is the concern for user-defined `ops`?

Yes, that was my concern. To be honest; perhaps that's a theoretical example, and not a big concern, but I wanted to _mention_ it at least, so that we can take it into account during review 😅